File: iterableapi-ui/src/main/java/com/iterable/iterableapi/ui/inbox/IterableInboxComparator.java
package com.iterable.iterableapi.ui.inbox;

import androidx.annotation.NonNull;

import com.iterable.iterableapi.IterableInAppMessage;

import java.util.Comparator;

/**
 * An interface to specify custom ordering of Inbox messages
 * See {@link Comparator}
 */
public interface IterableInboxComparator extends Comparator<IterableInAppMessage>{
    int compare(@NonNull IterableInAppMessage message1, @NonNull IterableInAppMessage message2);
}

File: iterableapi-ui/src/main/java/com/iterable/iterableapi/ui/inbox/IterableInboxComparator.kt
package com.iterable.iterableapi.ui.inbox

import androidx.annotation.NonNull

import com.iterable.iterableapi.IterableInAppMessage

import java.util.Comparator

/**
 * An interface to specify custom ordering of Inbox messages
 * See [Comparator]
 */
interface IterableInboxComparator : Comparator<IterableInAppMessage> {
    override fun compare(@NonNull message1: IterableInAppMessage, @NonNull message2: IterableInAppMessage): Int
}
